internal/bminventory/inventory.go

```diff
@@ -1577,7 +1577,7 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 		uerr := errors.Wrap(err, "Failed to register host: error creating host metadata")
```

Comment:
@RazRegev We assume here that hostApi.RegisterHost fails because the host already exists?

Reply:
We are not assuming it here, the check for that is done in transaction.go, here we just converting the error type to installer.NewRegisterHostForbidden() if it receive error code 403. 
By that agent will stop trying to register.